uniswap_swap_detection/src/agent.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  ethers,
+} from "forta-agent";
+
+
+import { UNISWAP_ADDRESS, SWAP_EVENT, FUNCTION_EXACT, SWAP_ROUTER, ABI } from "./constant";
+
+import { getPool } from "./utils";
+
+
+
+const provideHandleTransaction = () => async (txEvent: TransactionEvent) => { 
+
+
+  const findings: Finding[] = [];
+
+  const functionCalls =  txEvent.filterFunction([FUNCTION_EXACT], SWAP_ROUTER );
```

Comment:
Currently, the implementation is looking for instances where the `exactInputSingle` function is called on the Router contract. _Then_ proceed to check if there was a `swap` in that same transaction.

However, this wouldn't work as you hoped because the bot is supposed to detect instances of _any_ `swap` happening. The bot, in its current implementation would catch instances where a `swap` was preceded by a `exactInputSingle` call.

This is _exactly_ what block at L25-28 is saying to do.